### PR TITLE
ProductType and Suppliers endpoints

### PR DESCRIPTION
**Pull request by darmhoo**
#### What does this PR do?
- (replace me) ProductType and Suppliers endpoints

#### Description of Task proposed in this pull request?
- (replace me) Added controller, repository, migration, routes, model for ProductType and Suppliers

#### How should this be manually tested (Quality Assurance)?
- Explain how to manually test this feature. For example if changes were made in
the UI or in the API, explain where and if any specific access is needed.

#### What are the relevant pivotal tracker stories?
- [Story ID:  replace text with story id](paste Pivotal Tracker link here)

#### Any background context you want to add (Operations Impact)?
- (replace me) List the steps that must be taken for this PR to work.
E.g.: `npm install`, `Add "github_key" to environment variables, ...`

Be sure to include all systems that needs to be changed or which system is
affected by the change (Ex: Requires Styled Components to be installed and
configured in `babel.rc`, file to be updated).

#### What I have learned working on this feature: [If you don't put anything here, you are doing it wrong!]
- (replace me) I learned how to create a pull request template

### CODE REVIEW

File: app/Http/Controllers/SupplierController.php [hunk 1]
```diff
@@ -3,128 +3,76 @@
...
-    public function edit($id)
+    public function show(Supplier $supplier)
     {
-        $banks = Bank::all();
-        $form = Supplier::findOrFail($id);
-        return response()->json([
-            'form' => $form,
-            'banks' => $banks,
-        ]);
+        return $this->sendSuccess($supplier->toArray(), 'Brand retrieved successfully');
```
Comment (AltaraNg) [inline app/Http/Controllers/SupplierController.php:51]: Brands or Supplier?

File: app/Http/Requests/SupplierRequest.php [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+                return Supplier::rules();
+            }
+            case 'PUT': {
+                return Supplier::updateRules($this->supplier->id);
+            }
+            default:break;
+        }
+    }
+
+    public function show(Supplier $supplier)
```
Comment (AltaraNg) [inline app/Http/Requests/SupplierRequest.php:41]: whats this for?

File: app/Repositories/SupplierRepository.php [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+namespace App\Repositories;
+
+
+use App\Helper\ExtractRequestObject;
+use App\Helper\OrderObject;
+use App\Supplier;
+
+class SupplierRepository extends Repository
+{
+    use OrderObject, ExtractRequestObject;
```
Comment (AltaraNg) [inline app/Repositories/SupplierRepository.php:13]: Remove these unused imports

File: app/Http/Controllers/SupplierController.php (hunk 1 above)
Comment (darmhoo) [reply app/Http/Controllers/SupplierController.php:51]: Sorry. My bad

File: app/Http/Requests/SupplierRequest.php (hunk 2 above)
Comment (darmhoo) [reply app/Http/Requests/SupplierRequest.php:41]: Maybe the supplier details will like to be viewed.

File: app/Repositories/SupplierRepository.php (hunk 3 above)
Comment (darmhoo) [reply app/Repositories/SupplierRepository.php:13]: Ok. Will so that now

File: app/Http/Requests/SupplierRequest.php (hunk 2 above)
Comment (darmhoo) [reply app/Http/Requests/SupplierRequest.php:41]: Wrong placement sorry

**Pull request merged**